modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -47,34 +47,46 @@ enum class TraitAs: int
 using IEConfig = std::map<std::string, std::string>;
 
 namespace detail {
-    struct ParamDesc {
-        std::string model_path;
-        std::string weights_path;
-        std::string device_id;
-
-        // NB: Here order follows the `Net` API
-        std::vector<std::string> input_names;
-        std::vector<std::string> output_names;
-
-        using ConstInput = std::pair<cv::Mat, TraitAs>;
-        std::unordered_map<std::string, ConstInput> const_inputs;
-
-        // NB: nun_* may differ from topology's real input/output port numbers
-        // (e.g. topology's partial execution)
-        std::size_t num_in;  // How many inputs are defined in the operation
-        std::size_t num_out; // How many outputs are defined in the operation
-
-        enum class Kind { Load, Import };
-        Kind kind;
-        bool is_generic;
-        IEConfig config;
-
-        std::map<std::string, std::vector<std::size_t>> reshape_table;
-        std::unordered_set<std::string> layer_names_to_reshape;
-
-        // NB: Number of asyncrhonious infer requests
-        size_t nireq;
-    };
+/**
+* @brief This structure contains description of inference parameters
+* which is specific to OpenVINO models.
+*/
+struct ParamDesc {
```

Comment:
It must remain in the `detail`, just drop all the comments here -- this is not the structure users works with!!